src/main.rs

```diff
@@ -250,20 +252,60 @@ fn run_machete() -> anyhow::Result<bool> {
     Ok(has_unused_dependencies)
 }
 
-fn remove_dependencies(manifest: &str, dependencies_list: &[String]) -> anyhow::Result<String> {
+// handle a superset of all dependency name dashed/underscored variants: re'\w[-_]'
+fn dep_name_superset(dep_names: &[String]) -> HashSet<String> {
+    let mut unused: HashSet<String> = dep_names.iter().cloned().collect();
+    for dep in unused.clone() {
+        unused.insert(dep.replace('-', "_"));
+        unused.insert(dep.replace('_', "-"));
+    }
+    unused
+}
+
+fn remove_dependencies(manifest: &str, dependency_list: &[String]) -> anyhow::Result<String> {
     let mut manifest = toml_edit::Document::from_str(manifest)?;
-    let dependencies = manifest
+    let missing_table_msg = "unexpected missing table, please report with a test case on https://github.com/bnjbvr/cargo-machete";
+    let valid_tables = HashSet::from(["dependencies", "build-dependencies", "dev-dependencies"]);
+    let dependency_superset = dep_name_superset(dependency_list);
+
+    let mut dep_tables = manifest
         .iter_mut()
-        .find_map(|(k, v)| (v.is_table_like() && k == "dependencies").then_some(Some(v)))
-        .flatten()
-        .context("dependencies table is missing or empty")?
-        .as_table_mut()
-        .context("unexpected missing table, please report with a test case on https://github.com/bnjbvr/cargo-machete")?;
-
-    for k in dependencies_list {
-        dependencies
-            .remove(k)
-            .with_context(|| format!("Dependency {k} not found"))?;
+        // select for items that are tables with valid names
+        .filter(|(k, v)| v.is_table_like() && valid_tables.contains(k.display_repr().as_ref()))
+        .map(|(k, v)| {
+            let table = v.as_table_like_mut().context(missing_table_msg)?;
+            Ok((k, table))
+        })
+        .filter(|v| match v {
+            Ok((_, table)) if dependency_superset.iter().any(|k| table.contains_key(k)) => true,
+            Err(_) => true,
+            Ok(_) => false,
+        })
+        .collect::<Result<Vec<(KeyMut, &mut dyn TableLike)>, anyhow::Error>>()?;
+
+    for dep in dependency_list {
+        // for now
+        let mut removed_one = false;
+        for (name, table) in &mut dep_tables {
+            if table
+                .remove(dep)
+                .or_else(|| table.remove(dep.replace('_', "-").as_str()))
+                .is_some()
+            {
+                removed_one = true;
+                log::debug!("removed {name}.{dep}");
+            } else {
+                log::trace!("no match for {name}.{dep}");
+            }
+        }
+        if !removed_one {
+            let tables = dep_tables
+                .iter()
+                .map(|(k, _)| format!("{k}"))
+                .collect::<Vec<String>>()
+                .join(", ");
+            bail!(anyhow!("{dep} not found").context(format!("tables: {tables}")));
```

Comment:
`bail!` with a `context` doesn't seem very idiomatic to me: maybe we could instead plain use `bail!` here, and join the two error messages with a `\n` or `:`?